WordPress/openverse
# Add baseline functionality for running Playwright tests against live environments

## Fixes



Related to https://github.com/WordPress/openverse/issues/4706 by @sarayourfriend 

## Description


This PR introduces the same HMAC request signing strategy used for #4908 tests against live environments to use with Playwright. This allows Playwright tests to run against the live environment and bypass Cloudflare's WAF, otherwise it gets blocked as automated traffic.

To enable the signing transparently, I've created an override of the `page` fixture that automatically adds the HMAC headers to requests. This requires using `test` imported from `~~/test/playwright/utils/test` so the fixture is always available. I've updated all imports of `test` to use the new location and added an ESLint rule to encourage importing from the fixture-enabled location.

Additionally, I've added a new workflow file similar to the k6 workflow for running the Playwright smoke tests. The smoke tests are identified via the [test annotation](https://playwright.dev/docs/test-annotations#tag-tests) `@deployment-smoketest`. The workflow is dispatched after staging deployments, before the load test runs.

For this initial integration, I've just tagged some keyboard navigation tests with `@deployment-smoketest`, but to fully close the issue, more tests should be included.

## Testing Instructions

CI should pass, regular Playwright test runs should be unaffected.

You should be able to run the deployment smoke tests locally using the following command:

```bash
env HMAC_SIGNING_SECRET=<signing secret> PLAYWRIGHT_BASE_URL=https://staging.openverse.org just p frontend test:playwright:local --grep '@deployment-smoketest'
```

Maintainers can pull the signing secret as needed.

I also added temporary triggers for the new workflow file to run on pull_requests, and you can check out the successful run of that [at this link](https://github.com/WordPress/openverse/actions/runs/11510009898/job/32041101434?pr=5075). I've removed the pull_request trigger now, so it will only run when dispatched, just like the k6 workflow.

## Checklist


- [N/A] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
